CMakeLists.txt

```diff
@@ -70,6 +72,43 @@ macro(apply_llvm_compile_flags target)
     target_compile_definitions(${target} PRIVATE ${LLVM_DEFINITIONS})
 endmacro()
 
+# Check if using external LLVM/MLIR install tree
+if (DEFINED MLIR_DIR)
```

Comment:
We should really not need two flags. If someone is using an LLVM build without building MLIR project, CMake will throw an error and we should at that point inform user what to do.

Reply:
Perhaps I am not understanding it right. What does `MLIR_DIR` point to?

Reply:
MLIR_DIR is defined to guide "find_package(MLIR)" work correctly in CMake.
LLVM_DIR is for "find_package(LLVM)"
Both were already used in the code base. Not introduced by this PR.
Actually find_package(MLIR) will call find_package(LLVM) internally. So we can clean it up but that requires more changes.